check_env.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+# Standard library
+import os
+import json
+import inspect
+import sys
+import types
+import six
+
+from astropy import log as logger
+
+def check_environment(tutorial=None):
+    error=False
+    enter=False
+    warnings=False
+    logger.info("Running Environment Tests...")
+    _orig_path = os.getcwd()
+    tutorials_base = os.path.join(_orig_path,'tutorials')
+    for tutorial_name in os.listdir(tutorials_base) :
+        if (tutorial_name==tutorial or tutorial==None):
+            enter=True
+            tutorial_path = os.path.join(tutorials_base, tutorial_name)
+            os.chdir(tutorial_path)
+            try:
+                with open("requirements.json") as data_file:
+                    data=json.load(data_file)
+                    for pkgname in data:
+                        for subinfo in data[pkgname]:
+                            if subinfo=='min_version':
+                                if(not minversion(pkgname,data[pkgname][subinfo])):
+                                    logger.error("Package "+pkgname+" doesn't satisfy requirements of Tutorial: "+tutorial_name)
+                                    error=True
+                            elif subinfo=='pref_version':
+                                if(not minversion(pkgname,data[pkgname][subinfo])):
+                                    logger.warning("Please upgrade Package "+pkgname+" to version "+data[pkgname][subinfo]+" for Tutorial: "+tutorial_name)
+                                    warnings=True
+            except:
+                logger.error("Environment Check Failed ! requirements.json not found");
+
+    if enter==False:
+        logger.error("Wrong tutorial name entered !")
+        return
+    if warnings==True:
+        logger.warning("Please resolve the above warnings soon")
+    if error==False:
+        logger.info("Environment Check Passed")
+    else:
+        logger.info("Please resolve the above errors to continue!")
+
+    os.chdir(_orig_path)
+
+
+
+#ripped off from astropy.utils.introspection
```

Comment:
Better to be a little more formal and descriptive in comments and saying something like
```
# Functions resolve_name() and minversion() were copied from astropy.utils.introspection
```